nvflare/app_common/executors/task_script_runner.py

```diff
@@ -62,8 +62,13 @@ def get_sys_argv(self):
 
     def get_script_full_path(self, script_path) -> str:
         target_files = None
+        script_filename = os.path.basename(script_path)
+        script_dir = os.path.dirname(script_path)
         for r, dirs, files in os.walk(os.getcwd()):
-            target_files = [os.path.join(r, f) for f in files if f == script_path]
+            path_components = r.split(os.sep)
+            if script_dir == "" or script_dir in path_components:
+                target_files = [os.path.join(r, f) for f in files if f == script_filename]
```

Comment:
This if condition may not handle the path like this "src/sub/cli.py", and search for "src/cli.py" so well. Probably we can simply do this:

            target_files = [os.path.join(r, f) for f in files if os.path.join(r, f).endswith(script_path)]

Reply:
what I mean is if there are two files in the script, one is "src/cli.py" another one is "src/sub/cli.py", then the solution may find the wrong script to execute. Of course if there only one in "src/sub/cli.py", you can not find it by "src/cli.py".

Reply:
if the script is in "src/sub/cli.py", search for "src/cli.py" should result with not found, instead of True.